### ISSUE TITLE
[Bug]: Crawling has become worse for some urls compared to prevous versions

### ISSUE THREAD

**Issue by vikaskookna** [unclecode/crawl4ai#812]
### crawl4ai version

0.5.0

### Expected Behavior

@unclecode When i crawl a website it should give me all pages not just home page.

### Current Behavior

For some of the websites which earlier versions like 0.3.74 were able to crawl, are now not working on the new version of Crawl4AI. 

Can you please investigate this.

### Is this reproducible?

Yes

### Inputs Causing the Bug

```bash
https://buildfaith.org
```

### Steps to Reproduce

```bash
Just run the crawler to extract all urls of the link and it will just produce home page only.
```

### Code snippets

```python

```

### OS

macOS

### Python version

3.11

### Browser

Chrome

**Comment by bigbrother666sh** [comment]
also I Found
"https://arxiv.org/list/cs/recent",

V0.4.247 works well, but v0.4.248 can not crawl all content in the page

**Comment by unclecode** [comment]
@vikaskookna @bigbrother666sh thx for sharing, let me check. @aravindkarnam can you work on it? I still not clear what means "it returns only homepage"

**Comment by vikaskookna** [comment]
It means crawler just returns one page content which is home page rather than crawling all website urls in this particular case.

**Comment by aravindkarnam** [comment]
@vikaskookna I tried with older versions, still the site in question did not work. It's failing because there's a bot detection mechanism blocking us and giving a notice in html instead of actual page. It's more likely that the website recently implemented this mechanism, as we haven't changed any part of the code related to the scraping.

We are working on a workaround for this, though. It's currently being tested. I'll update this thread with a solution when that is released.

**Comment by bigbrother666sh** [comment]
hello @aravindkarnam  how about "https://arxiv.org/list/cs/recent"?

**Comment by aravindkarnam** [comment]
@bigbrother666sh For "https://arxiv.org/list/cs/recent", even the latest version worked(it fetched raw html). Maybe any specific functionality(markdown generation, extraction etc) is not working as expected. Could you elaborate on "can not crawl all content in the page". What's the missing content and is in raw html or markdown?

**Comment by bigbrother666sh** [comment]
hi  @aravindkarnam 

sorry for late reply

here's what I got the raw_html


`"html": "<!DOCTYPE html><html lang=\"en\"><head>\n    <meta charset=\"utf-8\">\n<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n<!-- new favicon config and versions by realfavicongenerator.net -->\n<link rel=\"apple-touch-icon\" sizes=\"180x180\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/apple-touch-icon.png\">\n<link rel=\"icon\" type=\"image/png\" sizes=\"32x32\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/favicon-32x32.png\">\n<link rel=\"icon\" type=\"image/png\" sizes=\"16x16\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/favicon-16x16.png\">\n<link rel=\"manifest\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/site.webmanifest\">\n<link rel=\"mask-icon\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/safari-pinned-tab.svg\" color=\"#b31b1b\">\n<link rel=\"shortcut icon\" href=\"https://static.arxiv.org/static/base/1.0.0a5/images/icons/favicon.ico\">\n<meta name=\"msapplication-TileColor\" content=\"#b31b1b\">\n<meta name=\"msapplication-config\" content=\"images/icons/browserconfig.xml\">\n<meta name=\"theme-color\" content=\"#b31b1b\">\n<!-- end favicon config -->\n<title>Search | arXiv e-print repository</title>\n<script defer=\"\" src=\"https://static.arxiv.org/static/base/1.0.0a5/fontawesome-free-5.11.2-web/js/all.js\"></script>\n<link rel=\"stylesheet\" href=\"https://static.arxiv.org/static/base/1.0.0a5/css/arxivstyle.css\">\n<script type=\"text/x-mathjax-config\">\n  MathJax.Hub.Config({\n    messageStyle: \"none\",\n    extensions: [\"tex2jax.js\"],\n    jax: [\"input/TeX\", \"output/HTML-CSS\"],\n    tex2jax: {\n      inlineMath: [ ['$','$'], [\"\\\\(\",\"\\\\)\"] ],\n      displayMath: [ ['$$','$$'], [\"\\\\[\",\"\\\\]\"] ],\n      processEscapes: true,\n      ignoreClass: '.*',\n      processClass: 'mathjax.*'\n    },\n    TeX: {\n        extensions: [\"AMSmath.js\", \"AMSsymbols.js\", \"noErrors.js\"],\n        noErrors: {\n          inlineDelimiters: [\"$\",\"$\"],\n          multiLine: false,\n          style: {\n            \"font-size\": \"normal\",\n            \"border\": \"\"\n          }\n        }\n    },\n    \"HTML-CSS\": { availableFonts: [\"TeX\"] }\n  });\n</script>\n<script src=\"//static.arxiv.org/MathJax-2.7.3/MathJax.js\"></script></head></html>",`

Obviously this is much less than the content of the page itself.

[e529eb.json](https://github.com/user-attachments/files/19271075/e529eb.json)

![Image](https://github.com/user-attachments/assets/d90f3e2c-c81b-46b6-957c-6a48c1d545ea)

In addition, I noticed that during the crawling stage, there was an error 

`[[ERROR]... × Error updating image dimensions: Page.evaluate: Execution context was destroyed, most likely because of a navigation]`

I found this phenomenon to occur in both 0.4.248 and 0.5.00.

**Comment by Ahmed-Tawfik94** [comment]
Hello @vikaskookna  @bigbrother666sh could you update to our latest release v0.7.7 and let us know if you are still facing this issue.

**Comment by unclecode** [comment]
Hi, thanks for taking the time to report this\! We're closing this issue as it was reported on an older version and hasn't had recent activity. If you're still experiencing this on the latest version (0.8.x), please feel free to reopen with updated reproduction steps. We're happy to take another look\!
